src/components/LibrariesList.tsx

```diff
@@ -0,0 +1,89 @@
+import * as React from "react";
+import { PanelGroup } from "react-bootstrap";
+import { Store, Reducer } from "redux";
+import { State } from "../reducers/index";
+import { LibrariesData, LibraryData } from "../interfaces";
+import { connect } from "react-redux";
+import ActionCreator from "../actions";
+import LibrariesListItem from "./LibrariesListItem";
+
+export interface LibrariesListStateProps {
+  libraries?: LibrariesData;
+}
+
+export interface LibrariesListOwnProps {
+  store?: Store<State>;
+  csrfToken?: string;
+}
+
+export interface LibrariesListDispatchProps {
+  fetchData: () => Promise<LibrariesData>;
+}
+
+export interface LibrariesListState {
+  activeKey: string;
+}
+
+export interface LibrariesListProps extends LibrariesListStateProps, LibrariesListOwnProps, LibrariesListDispatchProps {};
+
+export class LibrariesList extends React.Component<LibrariesListProps, LibrariesListState> {
+
+  constructor(props: LibrariesListProps) {
+    super(props);
+    this.state = { activeKey: "" };
+    this.isActive = this.isActive.bind(this);
+    this.select = this.select.bind(this);
+  }
+
+  isActive(library: LibraryData) {
+    return this.props.libraries.libraries.indexOf(library) === parseInt(this.state.activeKey);
+  }
+
+  select(idx: string) {
+    this.setState({ activeKey: idx});
+    this.props.fetchData();
+  }
+
+  render(): JSX.Element {
+    return(
+      <PanelGroup accordion activeKey={this.state.activeKey}>
+        { this.props.libraries &&
+          this.props.libraries.libraries.map(library =>
+            <LibrariesListItem
+              key={library.uuid}
+              idx={`${this.props.libraries.libraries.indexOf(library)}`}
```

Comment:
Does PanelGroup require this for the activeKey?

Reply:
Exactly!

Reply:
I'll look at it again, but I'm pretty sure...

Reply:
I looked at it again.  Yes, you need both `activeKey` and `expanded`.  Annoying, but not hurting anything.  (And it's much less annoying now that the panel doesn't need the `idx` prop!)